### PR TITLE
libnoaslr freebsd implementation proposal.

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

**Comment by domenukk** [comment]
Cool stuff, thank you!

File: utils/noaslr/libnoaslr/Cargo.toml [hunk 1]
```diff
@@ -14,3 +14,4 @@ crate-type = ["dylib"]
 anyhow = { version = "1.0.71", default-features = false }
 ctor = { version = "0.2.2", default-features = false }
 nix = { version = "0.26.2", default-features = false, features = ["process", "personality"] }
+libc = "0.2"
```
Comment (domenukk) [inline utils/noaslr/libnoaslr/Cargo.toml:17]: Should this be in
```toml
[target.'cfg(target_os = "freebsd")'.dependencies]
```

File: utils/noaslr/libnoaslr/src/lib.rs [hunk 2]
```diff
@@ -39,7 +44,65 @@ fn libnoaslr() -> Result<()> {
     Ok(())
 }
 
-#[cfg(any(target_os = "linux", target_os = "android"))]
+#[cfg(not(any(target_os = "linux", target_os = "android")))]
```
Comment (domenukk) [inline utils/noaslr/libnoaslr/src/lib.rs:47]: Just use `target_os = "freebsd"` here? Or could this also work on other unix systems?

**Review by domenukk — APPROVED** [review]

File: utils/noaslr/libnoaslr/Cargo.toml (hunk 1 above)
Comment (devnexen) [reply utils/noaslr/libnoaslr/Cargo.toml:17]: sure, only few oses down the road would apply anyway.

File: utils/noaslr/libnoaslr/src/lib.rs (hunk 2 above)
Comment (devnexen) [reply utils/noaslr/libnoaslr/src/lib.rs:47]: netbsd might work with few adaptations however make no sense for some (e.g. openbsd does not allow to disable aslr whatsoever) so I ll follow your advice.

**Pull request merged**